### PR TITLE
python311Packages.ydata-profiling: init at 4.7.0

### PR DESCRIPTION
**Pull request by katanallama**
## Description of changes

Adds ydata-profiling package and visions package to nixpkgs. 

### PhiK
Updated dependencies `scikit-build` to new `scikit-build-core`, added `pathspec` and `pyproject-metadata`. 

No longer need to disable the following tests, either removed or now pass
`"test_significance_matrix_hybrid"`
`"test_significance_matrix_mc"`
`"tests/phik_python/integration/"`

### Wordcloud
[Stopword](https://github.com/amueller/word_cloud/blob/main/wordcloud/stopwords) file added, was not previously included causing a build failure.

### ydata-profiling
Also tested with one of their [provided examples](https://github.com/ydataai/ydata-profiling/blob/master/examples/usaairquality/usaairquality.py) that uses both `phik` and `wordcloud` using:

```shell
nix-shell -I nixpkgs=. -p 'with import <nixpkgs> {}; [
  (python3.withPackages (ps: with ps; [ ydata-profiling ]))
]'
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python311Packages.phik</li>
    <li>python311Packages.phik.dist</li>
    <li>python311Packages.visions</li>
    <li>python311Packages.visions.dist</li>
    <li>python311Packages.wordcloud</li>
    <li>python311Packages.wordcloud.dist</li>
    <li>python311Packages.ydata-profiling</li>
    <li>python311Packages.ydata-profiling.dist</li>
    <li>python312Packages.visions</li>
    <li>python312Packages.visions.dist</li>
    <li>python312Packages.wordcloud</li>
    <li>python312Packages.wordcloud.dist</li>
  </ul>
</details>

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/visions/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    repo = "visions";
+    rev =
+      "5fe9dd0c2a5ada0162a005c880bac5296686a5aa"; # no tagged release for 0.7.6
+    hash = "sha256-SZzDXm+faAvrfSOT0fwwAf9IH7upNybwKxbjw1CrHj8=";
+  };
+
+  propagatedBuildInputs = [ attrs multimethod networkx numpy pandas ];
+
+  nativeCheckInputs = [ pytestCheckHook imagehash ]
+    ++ passthru.optional-dependencies.type-geometry
```
Comment (numinit) [inline pkgs/development/python-modules/visions/default.nix:37]: If you would like these passthrus to be overridable, do something like

```
buildPythonPackage (final: rec {
  # snip
  ++ final.passthru.optional-dependencies.type-geometry
  ++ final.passthru...
```

Comment (katanallama) [reply pkgs/development/python-modules/visions/default.nix]: Thanks, updated!

**Comment by katanallama** [comment]
Result of `nixpkgs-review pr 303350` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python311Packages.phik</li>
    <li>python311Packages.phik.dist</li>
    <li>python311Packages.visions</li>
    <li>python311Packages.visions.dist</li>
    <li>python311Packages.wordcloud</li>
    <li>python311Packages.wordcloud.dist</li>
    <li>python311Packages.ydata-profiling</li>
    <li>python311Packages.ydata-profiling.dist</li>
    <li>python312Packages.visions</li>
    <li>python312Packages.visions.dist</li>
    <li>python312Packages.wordcloud</li>
    <li>python312Packages.wordcloud.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/visions/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    repo = "visions";
+    rev = "5fe9dd0c2a5ada0162a005c880bac5296686a5aa"; # no tagged release for 0.7.6
+    hash = "sha256-SZzDXm+faAvrfSOT0fwwAf9IH7upNybwKxbjw1CrHj8=";
+  };
+
+  propagatedBuildInputs = [ attrs multimethod networkx numpy pandas ];
+
+  nativeCheckInputs =
+    [ pytestCheckHook imagehash ]
+    ++ lib.flatten (builtins.attrValues passthru.optional-dependencies);
```
Comment (numinit) [inline pkgs/development/python-modules/visions/default.nix:43]: same here, use final.passthru

**Review by numinit — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/visions/default.nix (hunk 2 above)
Comment (katanallama) [reply pkgs/development/python-modules/visions/default.nix:43]: I'm not sure this is the appropriate use of `final`, seems `lib.fix` is what's used under the hood and also what packages like [`sqlalchemy`](https://github.com/NixOS/nixpkgs/blob/274d22e4c269a3ebc4421529215f3b972db24c84/pkgs/development/python-modules/sqlalchemy/default.nix#L70) use.

Comment (numinit) [reply pkgs/development/python-modules/visions/default.nix:43]: Sounds good.

**Review by numinit — APPROVED** [review]

**Pull request closed without merging**